src/mito2/src/memtable/partition_tree/dict.rs

```diff
@@ -142,6 +142,7 @@ impl KeyDictBuilder {
             if let Some(sparse_key) = sparse_key {
                 pk_to_index_map.insert(sparse_key, i as PkIndex);
             }
+            pk_to_index_map.insert(full_pk, i as PkIndex);
```

Comment:
non-blocking: Not sure about the impact of this fix on memory usage. But we must ensure correctness first.